File: src/main/webapp/app/DOM/Theme.ts
import {Link} from "./link";
/**
 * Created by Jordan on 19/02/2016.
 */
export class Theme{
    themeId:number;
    themeName:string;
    description:string;
    organisation:string;
    iconUrl:string;
    links:Link[];

    constructor(themeId:number, themeName:string, description:string, organisation:string) {
        this.themeId = themeId;
        this.themeName = themeName;
        this.description=description;
        this.organisation=organisation;
    }

    static fromJson(json:any):Theme{
        var theme = new Theme(json.themeId,json.themeName,json.description,json.organisation);
        theme.iconUrl=json.iconUrl;

        for (var i = 0; i < json.links.length; i++){
            theme.links[i] = Link.fromJson(json.links[i])
        }

        return theme;
    }

    static createEmpty(): Theme {
        var theme = new Theme();
        return theme;
    }
}

File: src/main/java/be/kdg/kandoe/backend/dom/other/Theme.java
package be.kdg.kandoe.backend.dom.other;

import be.kdg.kandoe.backend.dom.game.Card;
import be.kdg.kandoe.backend.dom.users.User;
import org.springframework.hateoas.Identifiable;

import javax.persistence.*;
import java.io.Serializable;
import java.util.List;
import java.util.Set;

/**
 * Represents a theme sessions will be created about.
 * Cards can be added to a theme. Can have multiple subthemes.
 */
@Entity
public class Theme implements Serializable, Identifiable<Integer> {

    @Id
    @Column(name = "ThemeId", nullable = false)
    @GeneratedValue
    private Integer themeId;

    @Column(name = "ThemeName", nullable = false)
    private String themeName;

    @Column(name = "Description", nullable = false)
    private String description;

    @Column(name = "ThemeIcon")
    private String iconURL;

    @OneToMany(targetEntity = SubTheme.class, fetch = FetchType.EAGER)
    private Set<SubTheme> subThemes;

    @ManyToOne(targetEntity = Organisation.class)
    private Organisation organisation;

    @OneToMany(targetEntity = Card.class, fetch = FetchType.EAGER)
    private Set<Card> cards;

    @ManyToOne(targetEntity = User.class, fetch = FetchType.EAGER)
    private User creator;

    public Theme() {
    }

    public Theme(String themeName) {
        this.themeName=themeName;
    }

    public String getThemeName() {
        return themeName;
    }

    public void setThemeName(String themeName) {
        this.themeName = themeName;
    }

    public String getDescription() {
        return description;
    }

    public void setDescription(String description) {
        this.description = description;
    }

    public String getIconURL() {
        return iconURL;
    }

    public void setIconURL(String iconURL) {
        this.iconURL = iconURL;
    }

    public Set<SubTheme> getSubThemes() {
        return subThemes;
    }

    public void setSubThemes(Set<SubTheme> subThemes) {
        this.subThemes = subThemes;
    }

    public Organisation getOrganisation() {
        return organisation;
    }

    public void setOrganisation(Organisation organisation) {
        this.organisation = organisation;
    }

    public Set<Card> getCards() {
        return cards;
    }

    public void setCards(Set<Card> cards) {
        this.cards = cards;
    }

    public User getCreator() {
        return creator;
    }

    public void setCreator(User creator) {
        this.creator = creator;
    }

    @Override
    public Integer getId() {
        return themeId;
    }

    public Integer getThemeId() {
        return themeId;
    }

    public void setThemeId(Integer themeId) {
        this.themeId = themeId;
    }
}